perf(mangler): use shorter `InlineString`

overlookmotel opened the pull request:
`SymbolId` is a `u32`, so there are a maximum of `u32::MAX` symbols. Therefore the longest mangled symbol name is 6 characters (`xKrTKr` for `u32::MAX`).

Use an 8-byte `InlineString` for symbol names (instead of 16 bytes).

Also, enable the tests for `oxc_mangler` crate. The tests were previously failing, but we didn't notice because they were disabled.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9552?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9552** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9552?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9552?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9551** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9551?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9550** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9550?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9549** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9549?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9548** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9548?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9547** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9547?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9546** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9546?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9519** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9519?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9522** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9522?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9521** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9521?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9520** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9520?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9516** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9516?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9552).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel commented:
Damn. Only +0.2% perf improvement.

camchenry reviewed (approved):
Maybe a small improvement but still worth it I think.

Boshen commented:
### Merge activity

* **Mar 4, 9:04 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
perf(mangler): use shorter `InlineString` (#9552)

`SymbolId` is a `u32`, so there are a maximum of `u32::MAX` symbols. Therefore the longest mangled symbol name is 6 characters (`xKrTKr` for `u32::MAX`).

Use an 8-byte `InlineString` for symbol names (instead of 16 bytes).

Also, enable the tests for `oxc_mangler` crate. The tests were previously failing, but we didn't notice because they were disabled.
